numpy/f2py/crackfortran.py

```diff
@@ -832,7 +832,7 @@ def appenddecl(decl, decl2, force=1):
         decl = {}
     if not decl2:
         return decl
-    if decl is decl2:
+    if decl == decl2:
```

Comment:
Are we certain that `decl` will have an `__eq__` method, and that this is not intentionally checking for "is the exact same object"? I'm not familiar at all with `f2py`, but I wouldn't be surprised if that was the case...